graphql-lint/src/lib.rs

```diff
@@ -0,0 +1,654 @@
+                            field.directives().for_each(|directive_usage| {
+                                self.visit_directive_usage_field(&r#type, &field, &directive_usage);
+                            });
+                        });
+                    }
+                    TypeDefinition::Union(union) => {
+                        self.visit_union(union);
+                        union
+                            .directives()
+                            .for_each(|directive| self.visit_directive_usage(&r#type, &directive))
+                    }
+                    TypeDefinition::Enum(r#enum) => {
+                        self.visit_enum(r#enum);
+                        r#enum
+                            .directives()
+                            .for_each(|directive| self.visit_directive_usage(&r#type, &directive));
+                        r#enum.values().for_each(|value| {
+                            self.visit_enum_value(&r#type, &value);
+                            value.directives().for_each(|directive_usage| {
+                                self.visit_directive_usage_enum_value(r#enum, &value, &directive_usage);
+                            });
+                        });
+                    }
+                    TypeDefinition::InputObject(input_object) => {
+                        self.visit_input_object(input_object);
+                        input_object
+                            .directives()
+                            .for_each(|directive| self.visit_directive_usage(&r#type, &directive));
+                        input_object.fields().for_each(|input_value| {
+                            self.visit_input_value(&r#type, &input_value);
+                            input_value.directives().for_each(|directive_usage| {
+                                self.visit_directive_usage_input_value(input_object, &input_value, &directive_usage);
+                            });
+                        });
+                    }
+                };
+            }
+            Definition::Directive(directive) => {
+                self.visit_directive(&directive);
+                directive
+                    .arguments()
+                    .for_each(|argument| self.visit_directive_argument(&directive, &argument));
+            }
+        });
+
+        self.diagnostics
+    }
+
+    fn case_check(current: &'a str, case: Case) -> CaseMatch<'_> {
+        let fix = match case {
+            Case::Pascal => current.to_pascal_case(),
+            Case::ShoutySnake => current.to_shouty_snake_case(),
+            Case::Camel => current.to_lower_camel_case(),
+        };
+
+        if fix == current {
+            CaseMatch::Correct
+        } else {
+            CaseMatch::Incorrect { current, fix }
+        }
+    }
+
+    pub fn visit_field_argument(
+        &mut self,
+        parent_type: &TypeDefinition<'a>,
+        field: &FieldDefinition<'a>,
+        argument: &InputValueDefinition<'a>,
```

Comment:
Minor, but these types are all `Copy` so you could drop the references and have:

```suggestion
        parent_type: TypeDefinition<'a>,
        field: FieldDefinition<'a>,
        argument: InputValueDefinition<'a>,
```

Reply:
Same with all the other functions that take cynic-parser types

Reply:
You also might be able to drop `'a` and just use `'_` (although without trying it I couldn't be sure 😆)

Reply:
Done